### DISCUSSION TITLE
Some suggestions

### DISCUSSION THREAD

**Discussion by shadowe1ite** [MissingCore/Music#42] [Ideas]
1. Option to Black list directories 
2. Fetch Lyrics from internet 
3. Spotify like lyrics sync (if possible to implement)
4. Add option to select multiple music with long press  to perform actions like add to playlist, delete, etc
5. Option to see details 
 ``` 
File path:
File name:
Size:
Last modified:
Format: 
Length:
Bit rate:
Sampling rate:
```
6. And option to edit metadata like album name, etc

Hope you will consider my suggestions 
Thank you

**Comment by The-Ladle** [comment]
I agree with the lyrics thing, but the ability to add a custom JSON for local synced lyrics would be amazing if possible.

**Comment by cyanChill** [comment]
I just wrote a new document; [`App Goals & Ideology`](https://github.com/MissingCore/Music/blob/main/docs/app-goals-ideology.md) which goes a bit on what I want this app to be along with things I don't want to add.

1. Just added with `v1.0.0-rc.11`.
2. Probably not due to cost & legal stuff (lyrics are considered Intellectual Property).
3. Same as 2.
4. I plan to add some sort of gesture/multiselect logic in the future for this app (no estimate time frame).
5. For additional details, this might get implemented whenever I update the modals for the tracks.
6. This would kill my sanity probably, you're probably better off using dedicated apps for this.

**Reply by ghost** [reply]
Could it work with song lyrics apis where you could put in your own api key for song lyrics
Because there are free lyric api services.

**Reply by cyanChill** [reply]
There's a couple of problems:
- Does the API actually return the lyrics in the response, or just a link to a page with them.
- Would this violate their TOS or Privacy Policy.
- Is their API even legal?

And there's the question on whether Google will allow this app on the Play Store.

**Reply by ghost** [reply]
Try this Api 100 queries/day https://www.lyrics.com/api.php

**Reply by cyanChill** [reply]
That API only returns a link to the page containing the lyrics; and their TOS probably prohibit our use case.

**Comment by cyanChill** [comment]
If you're using the APK version of the app, we released a preview of the next major version through the [`v2.0.0-rc.1`](https://github.com/MissingCore/Music/releases/tag/v2.0.0-rc.1) release, which adds some iteration of the remaining 2 points (4 & 5).
- For the long-press gesture to add or remove a track from a playlist, we instead opted to improving the general adding process. We added new UI when creating/editing a playlist which allows you to search for tracks or albums and directly add them to the playlist. With deleting, we made the tracks swipeable, with the swipe revealing a remove button which should make it slightly faster to remove tracks.
- With the additional track detail, we display those details as part of the redesign track modal.
